core/utils/ethabi.go

```diff
@@ -24,36 +24,29 @@ const (
 )
 
 // ConcatBytes appends a bunch of byte arrays into a single byte array
-func ConcatBytes(bufs ...[]byte) ([]byte, error) {
-	buffer := bytes.NewBuffer([]byte{})
-	for _, b := range bufs {
-		_, err := buffer.Write(b)
-		if err != nil {
-			return nil, err
-		}
-	}
-	return buffer.Bytes(), nil
+func ConcatBytes(bufs ...[]byte) []byte {
+	return bytes.Join(bufs, []byte{})
 }
 
 // EVMTranscodeBytes converts a json input to an EVM bytes array
 func EVMTranscodeBytes(value gjson.Result) ([]byte, error) {
 	switch value.Type {
 	case gjson.String:
-		return EVMEncodeBytes([]byte(value.Str))
+		return EVMEncodeBytes([]byte(value.Str)), nil
 
 	case gjson.False:
-		return EVMEncodeBytes(EVMWordUint64(0))
+		return EVMEncodeBytes(EVMWordUint64(0)), nil
 
 	case gjson.True:
-		return EVMEncodeBytes(EVMWordUint64(1))
+		return EVMEncodeBytes(EVMWordUint64(1)), nil
 
 	case gjson.Number:
 		word, err := EVMWordSignedBigInt(big.NewInt(int64(value.Num)))
 		if err != nil {
 			return []byte{}, nil
 		}
 
-		return EVMEncodeBytes(word)
+		return EVMEncodeBytes(word), nil
```

Comment:
I would separate this change from the ConcatBytes change.

Reply:
Thanks, I split these.